lib/ansible/modules/monitoring/grafana_datasource.py

```diff
@@ -310,7 +311,7 @@ def grafana_headers(module, data):
 def grafana_datasource_exists(module, grafana_url, name, headers):
     datasource_exists = False
     ds = {}
-    r, info = fetch_url(module, '%s/api/datasources/name/%s' % (grafana_url, name), headers=headers, method='GET')
+    r, info = fetch_url(module, '%s/api/datasources/name/%s' % (grafana_url, urlencode(name)), headers=headers, method='GET')
```

Comment:
urlencode take a dict as parameter.
you should replace this with `urlencode({"name": name})`

Reply:
I use quote_plus now